### PR TITLE
Frontend work for the student course view

### PR DESCRIPTION
**Pull request by fredy20501**

### CODE REVIEW

**Review by ravenmills3 — APPROVED** [review]
good job

**Comment by fredy20501** [comment]
Nobody has given a second review after a week. Since Tash and I have worked together on this one I'll count that as a review and merge it now. (we really need to merge this in and make sure it works for the demo since this is one of the major features we absolutely need)

**Pull request merged**
